cardano-db-sync/src/Cardano/DbSync/Epoch.hs

```diff
@@ -228,69 +232,51 @@ calculateEpochUsingCache cBlock cEpoch slotDetails = do
 -- We then the epoch in cache and add these new values to it, thus giving us a new updated epoch.
-calculateEpochGeneric :: Generic.Block -> DB.Epoch -> SlotDetails -> ExceptT SyncNodeError (ReaderT SqlBackend m) DB.Epoch
+calculateEpochGeneric ::
+  (MonadBaseControl IO m, MonadIO m) =>
+  Generic.Block ->
+  DB.Epoch ->
+  SlotDetails ->
+  ExceptT SyncNodeError (ReaderT SqlBackend m) DB.Epoch
 calculateEpochGeneric block cEpoch slotDetails = do
-  let newEpochTxCount = fromIntegral DB.epochTxCount cEpoch + length (Generic.blkTxs block)
-      newBlkCount = fromIntegral DB.epochBlkCount cEpoch + 1
+  let newEpochTxCount = fromIntegral (DB.epochTxCount cEpoch) + length (Generic.blkTxs block)
+      newBlkCount = fromIntegral $ DB.epochBlkCount cEpoch + 1
       txs = Generic.blkTxs block
-      outSum = sum $ map (unCoin . Generic.txOutSum) txs
+      outSum = sum $ map (fromIntegral . unCoin . Generic.txOutSum) txs
 
-  -- TODO: Vince- unable to work out how I get the `inSum` properlly, this is mostly coppied from
-  --       insertTx inside of Cardano.DbSync.Era.Shelley.Insert
-  newFees <- foldM (\gp transaction -> calNewFees transaction outSum gp) mempty (Generic.blkTxs block)
+  newFees <- mapM (`calNewFees` outSum) txs
+  let sumNewFees = sum newFees
 
-  pure DB.Epoch
-    { DB.epochOutSum = fromIntegral (outSum + fromIntegral DB.epochOutSum cEpoch)
-    , DB.epochFees = DB.DbLovelace $ fromIntegral DB.epochFees cEpoch -- + just need to add new fees of block
-    , DB.epochTxCount = fromIntegral newEpochTxCount
-    , DB.epochBlkCount = fromIntegral newBlkCount
-    , DB.epochNo = unEpochNo (sdEpochNo slotDetails)
-    , DB.epochStartTime = DB.epochStartTime cEpoch
-    , DB.epochEndTime = sdSlotTime slotDetails
-    }
+  pure
+    DB.Epoch
+      { DB.epochOutSum = fromIntegral (outSum + fromIntegral (DB.epochOutSum cEpoch))
+      , DB.epochFees = DB.DbLovelace (DB.unDbLovelace (DB.epochFees cEpoch) + sumNewFees)
+      , DB.epochTxCount = fromIntegral newEpochTxCount
+      , DB.epochBlkCount = fromIntegral newBlkCount
+      , DB.epochNo = unEpochNo (sdEpochNo slotDetails)
+      , DB.epochStartTime = DB.epochStartTime cEpoch
+      , DB.epochEndTime = sdSlotTime slotDetails
+      }
   where
-    calNewFees :: Generic.Tx -> Integer -> BlockGroupedData -> ExceptT SyncNodeError (ReaderT SqlBackend m) Word64
-    calNewFees tx outS grouped = do
-      resolvedInputs <- mapM (resolveTxInputs (fst <$> groupedTxOut grouped)) (Generic.txInputs tx)
+    -- calculating the fees for current block using Tx's
+    calNewFees ::
+      (MonadBaseControl IO m, MonadIO m) =>
+      Generic.Tx ->
+      Word64 ->
+      ExceptT SyncNodeError (ReaderT SqlBackend m) Word64
+    calNewFees tx outS = do
+      -- taking all the txInputs and getting their values from a DB lookup.
+      -- Errors are ignored but might actually need to handle this gracefully
+      qResInputRes <- lift $ mapM queryResolveInput (Generic.txInputs tx)
```

Comment:
This aproach goes against the incentive of reducing the db queries necessary. We already resolve the inputs in `insertTx` and calculate the tx fees. Isn't it possible to resuse these results?